design/retry-patching-configuration_design.md

```diff
@@ -0,0 +1,104 @@
+# Backup Restore Status Patch Retrying Configuration
+
+## Abstract
+When a restore completes, we want to ensure that the restore custom resource progress to the correct status.
+If a patch call fails to update status to completion, it should be retried up to a certain time limit.
+
+This design proposes a way to configure timeout for this retry time limit.
+
+## Background
+Original Issue: https://github.com/vmware-tanzu/velero/issues/7207
+
+Velero was performing a restore when the API server was rolling out to a new version.
+It had trouble connecting to the API server, but eventually, the restore was successful.
+However, since the API server was still in the middle of rolling out, Velero failed to update the restore CR status and gave up.
+
+After the connection was restored, it didn't attempt to update, causing the restore CR to be stuck at "In progress" indefinitely.
+This can lead to incorrect decisions for other components that rely on the backup/restore CR status to determine completion.
+
+## Goals
+- Make timeout configurable for retry patching by reusing existing [`--resource-timeout` server flag](https://github.com/vmware-tanzu/velero/blob/d9ca14747925630664c9e4f85a682b5fc356806d/pkg/cmd/server/server.go#L245)
+
+## Non Goals
+- Create a new timeout flag
+- Refactor backup/restore workflow
+
+
+## High-Level Design
+We will add retries with timeout to existing patch calls that moves a backup/restore from InProgress to a final status such as
+- Completed
+- PartiallyFailed
+- Failed
```

Comment:
I'm not sure there are cases here where the transition is to Failed or FailedValidation -- FailedValidation -- in those cases, I think we're mostly moving from New, so a future reconcile can just handle it normally. Although maybe retry is still appropriate there?

Reply:
Future reconcile isn't generally handled. So we should still retry or it can be stuck in a phase other than new. New or "" is the only phase being retried via future reconcile.
```go
	switch restore.Status.Phase {
	case "", api.RestorePhaseNew:
		// only process new restores
	default:
		r.logger.WithFields(logrus.Fields{
			"restore": kubeutil.NamespaceAndName(restore),
			"phase":   restore.Status.Phase,
		}).Debug("Restore is not handled")
		return ctrl.Result{}, nil
	}
```

Reply:
@kaovilai well the various WaitingForPluginOperations will also be retried on future reconcile, since the backup/restore operations controller re-reconcile to check progress on a periodic basis rather than on a cr-updated basis.

Reply:
But those aren't handled by the backup/restore controller, which only handles new CRs.